policy-controller/src/profiling.rs

```
@@ -0,0 +1,136 @@
+                            profile.write_to_vec(&mut file).unwrap();
+                            Response::builder()
+                                .header("content-type", "application/octet-stream")
+                                .header("content-disposition", "attachment; filename=profile.pb")
+                                .body(file)
+                                .unwrap()
+                        } else if obj.format == "folded" {
+                            let my_report = MyReport(report);
+                            let mut file = Vec::new();
+                            my_report.folded(&mut file).unwrap();
+                            Response::builder()
+                                .header("content-type", "text/plain")
+                                .header("content-disposition", "attachment; filename=folded")
+                                .body(file)
+                                .unwrap()
+                        } else if obj.format == "tracing-flamegraph" {
+                            use std::io::Write;
+                            let flush_guard = flame_layer.flush_on_drop();
+                            let _flush = flush_guard.flush();
+                            // let file = tracing_flame_buffer.clone();
+                            let mut file = Vec::new();
+                            file.write_all(&tracing_flame_buffer).expect("Failed to set the default subscriber");
+                            Response::builder()
+                                .header("content-type", "text/plain")
+                                .header("content-disposition", "attachment; filename=tfolded")
+                                .body(file)
+                                .unwrap()
+                        } else {
+                            Response::builder()
+                                .body(Vec::from("unknown value for format"))
+                                .unwrap()
```

This should return a 404 error